src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -388,6 +391,48 @@ public Long call() throws Exception {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public List<BitbucketServerTag> getTags() throws IOException, InterruptedException {
+        String url = String.format(API_TAGS_PATH, getUserCentricOwner(), repositoryName, 0, DEFAULT_PAGE_LIMIT);
+
+        try {
+            List<BitbucketServerTag> tags = new ArrayList<>();
+            String response = getRequest(url);
+            BitbucketServerTags page = JsonParser.toJava(response, BitbucketServerTags.class);
+            tags.addAll(page.getValues());
+            while (!page.isLastPage()) {
+                if (Thread.interrupted()) {
+                    throw new InterruptedException();
+                }
+                Integer limit = page.getLimit();
+                url = String.format(API_TAGS_PATH, getUserCentricOwner(), repositoryName, page.getNextPageStart(),
```

Comment:
encoding of url stuff... or build on top of the PR for moving to URITemplates from damnhandy

Reply:
#80 is the PR that should make all this easier